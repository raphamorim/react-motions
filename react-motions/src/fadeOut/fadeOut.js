import React from 'react'
import PropTypes from 'prop-types'
import styles from './styles'

function FadeOut({ children, duration, fillMode, infinite }) {
  const style = {
    animation: {
      WebkitAnimationName: "fadeOut",
      animationName: "fadeOut",
      WebkitAnimationDuration: `${duration}s`,
      animationDuration: `${duration}s`,
      WebkitAnimationFillMode: fillMode,
      animationFillMode: fillMode
    },
    infinite: {
      WebkitAnimationIterationCount: "infinite",
      animationIterationCount: "infinite",
    }
  }

  const styleComponent = infinite
    ? Object.assign(style.animation, style.infinite)
    : style.animation

  return (
    <div style={styleComponent}>
      {children}
    </div>
  )
}

FadeOut.propTypes = {
  children: PropTypes.node.isRequired,
  duration: PropTypes.number,
  fillMode: PropTypes.string,
  infinite: PropTypes.bool
}

FadeOut.defaultProps = {
  duration: 1,
  fillMode: "both",
  infinite: false
}

export default FadeOut
