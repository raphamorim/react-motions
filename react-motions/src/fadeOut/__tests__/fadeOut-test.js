import React from 'react';
import { mount } from 'enzyme';
import FadeOut from '../fadeOut';

describe('FadeOut', () => {
  it('should render', () => {
    const wrapper = shallow(<FadeOut>FadeOut</FadeOut>);

    expect(wrapper).toMatchSnapshot();
  });
  it('should receive duration prop', () => {
    const wrapper = mount(<FadeOut duration={3}>FadeOut</FadeOut>);

    expect(wrapper.props().duration).toEqual(3);
  });
  it('should receive fillMode prop', () => {
    const wrapper = mount(<FadeOut fillMode="both">FadeOut</FadeOut>);
    expect(wrapper.props().fillMode).toEqual("both");
  });
  it('should render correctly when infinite', () => {
    const wrapper = mount(<FadeOut infinite>FadeOut</FadeOut>);

    expect(wrapper.props().infinite).toBe(true);
  });
});
