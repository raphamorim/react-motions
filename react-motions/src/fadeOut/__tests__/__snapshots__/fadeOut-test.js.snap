// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FadeOut should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FadeOut
    duration={1}
    fillMode="both"
    infinite={false}
  >
    FadeOut
  </FadeOut>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "FadeOut",
      "style": Object {
        "WebkitAnimationDuration": "1s",
        "WebkitAnimationFillMode": "both",
        "WebkitAnimationName": "fadeOut",
        "animationDuration": "1s",
        "animationFillMode": "both",
        "animationName": "fadeOut",
      },
    },
    "ref": null,
    "rendered": "FadeOut",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "FadeOut",
        "style": Object {
          "WebkitAnimationDuration": "1s",
          "WebkitAnimationFillMode": "both",
          "WebkitAnimationName": "fadeOut",
          "animationDuration": "1s",
          "animationFillMode": "both",
          "animationName": "fadeOut",
        },
      },
      "ref": null,
      "rendered": "FadeOut",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
